src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPASITY = 10;
+    private static final int NOT_FOUND = -1;
+    private int capasity;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        capasity = DEFAULT_CAPASITY;
+        array = (T[]) new Object[DEFAULT_CAPASITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capasity) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (invalidIndexToAdd(index)) {
+            throw new ArrayListIndexOutOfBoundsException(index + " is out of range index");
+        }
+        if (size == capasity) {
+            grow();
+        }
```

Comment:
you can use the regular check if you place it after this `if`

Reply:
Do you mean this one grow() {
if (size == capasity) {

}
...
}